File: business/src/main/java/com/radioteria/business/services/user/api/UserService.java
package com.radioteria.business.services.user.api;

import java.util.Optional;

public interface UserService {
    Optional<User> findByEmail(String email);
    Optional<User> findById(Long id);

    boolean isPasswordMatch(String plainPassword, User user);
    boolean isEmailAvailable(String email);

    void register(String email, String plainPassword, String name);
    void activateByEmail(String email);
    void changePassword(User user, String newPlainPassword);
    void deactivate(User user);
    void delete(User user);
}

File: business/src/main/kotlin/com/radioteria/services/user/UserService.kt
package com.radioteria.services.user

import com.radioteria.db.entities.User

import java.util.Optional

interface UserService {
    fun isPasswordMatch(plainPassword: String, user: User): Boolean
    fun isEmailAvailable(email: String): Boolean

    fun register(email: String, plainPassword: String, name: String)
    fun activateByEmail(email: String)
    fun changePassword(user: User, newPlainPassword: String)

    fun findByEmail(email: String): User?
    fun findById(id: Long): User?

    fun deactivate(user: User)
    fun delete(user: User)
}
